### ISSUE TITLE
property-case

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#1029]
[Properties are case insensitive, but *custom properties aren't*.](https://www.w3.org/TR/css-syntax-3/#style-rules) We'd probably want to ignore non-standard variables too.

```css
/* property-case: "upper" */
a {
  DISPLAY: block;
  --custom-PropertyName: red;
}
```

- Name: `property-case`
- Primary options: `"upper"|"lower"`
- Secondary options: none
- Message: `Expected "${actual}" to be "${expected}" (property-case)`.

**Comment by evilebottnawi** [comment]
@jeddy3 i think for variables sensitive (or may be `variable-pattern`) we should create separate rule. What do you think?

**Comment by jeddy3** [comment]
Yes, this rule should only check the case of standard properties like `display`.

There is already a rule for specifying a pattern for custom properties: [`custom-property-pattern`](https://github.com/stylelint/stylelint/tree/master/src/rules/custom-property-pattern).

And plugins should be used for non-standard variables e.g. `stylelint-at-variable-pattern` (for less), `stylelint-dollar-variable-pattern` (for scss and precss) etc... I believe there's [a discussion](https://github.com/kristerkari/stylelint-scss/issues/8) over at `stylelint-scss` regarding the latter.

**Comment by evilebottnawi** [comment]
@jeddy3 Merged. Close.
